ui/components/multichain/pages/review-permissions-page/review-permissions-page.tsx

```diff
@@ -111,22 +120,48 @@ export const ReviewPermissions = () => {
     dispatch(hidePermittedNetworkToast());
   };
 
-  const networkConfigurations = useSelector(getNetworkConfigurationsByChainId);
+  const networkConfigurationsByCaipChainId = useSelector(
+    getAllNetworkConfigurationsByCaipChainId,
+  );
+
   const [nonTestNetworks, testNetworks] = useMemo(
     () =>
-      Object.entries(networkConfigurations).reduce(
+      Object.entries(networkConfigurationsByCaipChainId).reduce(
         ([nonTestNetworksList, testNetworksList], [chainId, network]) => {
-          const isTest = (TEST_CHAINS as string[]).includes(chainId);
-          (isTest ? testNetworksList : nonTestNetworksList).push(network);
+          const caipChainId = chainId as CaipChainId;
+          const isTestNetwork =
+            CAIP_FORMATTED_EVM_TEST_CHAINS.includes(caipChainId);
+          (isTestNetwork ? testNetworksList : nonTestNetworksList).push({
+            ...network,
+            caipChainId,
+          });
           return [nonTestNetworksList, testNetworksList];
         },
-        [[] as NetworkConfiguration[], [] as NetworkConfiguration[]],
+        [
+          [] as ((NetworkConfiguration | MultichainNetworkConfiguration) & {
+            caipChainId: CaipChainId;
+          })[],
+          [] as ((NetworkConfiguration | MultichainNetworkConfiguration) & {
+            caipChainId: CaipChainId;
+          })[],
+        ],
       ),
-    [networkConfigurations],
+    [networkConfigurationsByCaipChainId],
+  );
+
+  const _connectedChainIds = useSelector((state) =>
+    getAllPermittedChainsForSelectedTab(state, activeTabOrigin),
+  ) as CaipChainId[];
+
+  const connectedChainIds = _connectedChainIds.filter(
+    (chainId: InternalScopeString) => {
+      if (chainId === KnownCaipNamespace.Wallet) {
+        return false;
+      }
```

Comment:
try catch instead

Reply:
connect page has an occurence of this too

Reply:
working on this

Reply:
[308911a](https://github.com/MetaMask/metamask-extension/pull/29887/commits/308911ab0990a99760609a4023da88fec0b9f6f6)